### DISCUSSION TITLE
Show options "Type", "Title" and "Type - Title" in metadata editor structure panel

### DISCUSSION THREAD

**Discussion by solth** [kitodo/kitodo-production#4925] [Ideas]
#4757 added a switch that allows the user to change the label of the tree nodes in the structure tree from type to specific metadata values configured as `metsEditor.titleMetadata` in the central config file `kitodo_config.properties`. 

However, the menu currently displays the `id` of metadata defined in the ruleset (or rather just the values separated by comma after `metsEditor.titleMetadata` from the config file), which is not optimal, since normally the `label` of a metadata is used in the Kitodo metadata editor UI and the user is probably not familiar with the internal IDs of the metadata as defined in the ruleset.
<img width="289" alt="Bildschirmfoto 2021-12-27 um 10 02 54" src="https://user-images.githubusercontent.com/19183925/147455580-dadd7638-b720-4fbf-a078-31df0428f8f0.png">

~~It would be better to show the `label` of the corresponding metadata in the menu instead, since the user is familiar with that, optionally preceded with a "Metadata: ", to make the difference to "Type" even easier to understand.~~

After thinking this through with @IkramMaalej, I think the best solution would be the following:
- add new FunctionalMetadata for titles that can be configured with `use=title` or `use=structureTitle` in corresponding rulesets (credits go to @BartChris for the suggestion)
- remove the configuration key `metsEditor.titleMetadata` from `kitodo_config.properties`, since it doesn't respect the ruleset in use by the current process
- always offer the same three options in the menu in the metadata editors structure panel: "Type", "Title" and "Type - Title" instead of displaying multiple metadata names. This has the advantages that 1) the menu is easiar to understand because it always holds the same three options  2) different metadata for different structure elements can be displayed at the same time (for example "title" for monographs and "chapterTitle" for chapters, if the ruleset is configured like this) and 3) the user does not have to have intricate knowledge of the ruleset configuration when choosing what information he wants to see displayed on the structure tree nodes.

For the options "Title" and "Type - Title" only the metadata in the current process has to be checked when searching for structure title information to display. 

@subhhwendt , @andre-hohmann could you give your opinion on this idea?

**Comment by IkramMaalej** [comment]
@solth Can you please check if the 'Language for metadata' of the user used for this test is English? 

 If the user's language for metadata is English and no English label is configured for the metadata, the metadata ID will be used as label.
 `<key id="vrzng_enht_titel">
    <label lang="de">Titel</label>
</key>`

**Comment by BartChris** [comment]
This is an useful feature. But may i ask why this has to be configured using the kitodo_config.properties? I would prefer if it is possible to define for each division in the ruleset, which key should be used for the display of the label. Would that be an option?

**Comment by solth** [comment]
@IkramMaalej it seems the label was not displayed because the whole metadata was not defined in the process' ruleset. So the string was taken directly from the `metsEditor.titleMetadata` configuration key in the `kitodo_config.properties` file and was not really the ID of some defined metadata. 

I think instead, only those entries in the configuration that represent metadata of the current process' ruleset should be used when searching for title metadata! As we discussed,  the menu should also probably not include the individual metadata entries that are eligable as title, but instead just offer the options "Type", "Title" and "Type - Title", so that multiple metadata keys for different structure types can be displayed at the same time and the user also does not have to worry which metadata acts as title for which structural element. 

@BartChris I think the reason why this is currently configured in the `kitodo_config.properties` file and not in the ruleset is just that I didn't think of it back then when this was implemented. I agree with you that configuring this via the ruleset would be the better solution and would also be more in line with other "functional" metadata configurations like "higherLevelIdentifier" etc.

I will convert this issue into a discussion because further evaluation showed the original issue description is not really the best solution for this.

**Comment by solth** [comment]
@andre-hohmann and @subhhwendt could you weigh in on this issue?

**Reply by subhhwendt** [reply]
I agree that moving the configuration of this useful switch should be part of the ruleset!! Considerung all details in "kitodo_config.properties is not my daily approach. I would prefer "use=structureTitle" as attribute. And the final choice "Type", "Title" and "Type - Title" will be clear to all users.

**Reply by andre-hohmann** [reply]
I can only agree. Using the ruleset is better than `kitodo_config.properties`. 
I suggest `use=structureTreeTitle`, if it is possible - it may help to derive the function of the attribute. If it is not possible, `use=structureTitle `should be applied. 
The labels "Type", "Title" and "Type - Title" is perfect. 

Is this also applied to the linked volumes of subordinates processes? For example, the elements which represents the volumes of a periodical?

**Reply by andre-hohmann** [reply]
I want to point out that the configuration of the ruleset is time consuming and should be done from the beginning of the rulesets creation. 
Thus, the default ruleset of Kitodo.Production must be customized according to the final decision regarding labeling, ... of the attribute.

**Reply by BartChris** [reply]
"Is this also applied to the linked volumes of subordinates processes? For example, the elements which represents the volumes of a periodical?"

Just in addition to that as i just checked again: The display for the linked child processes of a parent in the metadata editor is currently defined statically by concatinating the document type, the process ID and the process title.

https://github.com/kitodo/kitodo-production/blob/e2403b6418b543dccb5464c1d39078febf028fc7/Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java#L569-L570

One option is to use the displaySummary here, as it is done in the TitleRecordLink dialog while importing. Here a customized string (`tooltip`) is created from the fields which are marked as `displaySummary`:

https://github.com/kitodo/kitodo-production/blob/e2403b6418b543dccb5464c1d39078febf028fc7/Kitodo/src/main/java/org/kitodo/production/forms/createprocess/TitleRecordLinkTab.java#L247

https://github.com/kitodo/kitodo-production/blob/e2403b6418b543dccb5464c1d39078febf028fc7/Kitodo/src/main/webapp/WEB-INF/templates/includes/processFromTemplate/titleRecordLink.xhtml#L87-L88

This also brings up the question wether we need `use=structureTreeTitle` if we already have `use=displaySummary`, but maybe both serve different purposes.

Edit: I suppose `displaySummary` can be used for representing processes and `use=structureTreeTitle` for any division. Then it would be justified to have both.

**Comment by matthias-ronge** [comment]
Yes, the ruleset’s `use=` attribute seems a well place for this. Maybe, a `use="structureLabel"` will add it to the select menu. For example, you add the `use="structureLabel"` to the `<key>`s _title_ and _count_, then you will get options in the menu:

- type
- type - tile
- title
- type - count
- count

**Comment by andre-hohmann** [comment]
See also discussion in 
- #5016 

Especially the comment https://github.com/kitodo/kitodo-production/issues/5016#issuecomment-1380087772.
